Brokerages/Oanda/OandaRestApiBase.cs

```diff
@@ -347,110 +366,35 @@ public void SetJob(LiveNodePacket job)
-                                          select symbol).ToList();
-                if (symbolsToSubscribe.Count == 0)
-                    return;
-
-                Log.Trace("OandaBrokerage.Subscribe(): {0}", string.Join(",", symbolsToSubscribe.Select(x => x.Value)));
-
-                // Oanda does not allow more than a few rate streaming sessions,
-                // so we only use a single session for all currently subscribed symbols
-                symbolsToSubscribe = symbolsToSubscribe.Union(SubscribedSymbols.ToList()).ToList();
-
-                SubscribedSymbols = symbolsToSubscribe.ToHashSet();
-
-                ProcessSubscriptionRequest();
-            }
-        }
-
         /// <summary>
         /// Removes the specified configuration
         /// </summary>
         /// <param name="dataConfig">Subscription config to be removed</param>
         public void Unsubscribe(SubscriptionDataConfig dataConfig)
         {
-            Unsubscribe(new Symbol[] { dataConfig.Symbol });
+            _subscriptionManager.Unsubscribe(dataConfig);
             Aggregator.Remove(dataConfig);
         }
 
-        /// <summary>
-        /// Removes the specified symbols from the subscription
-        /// </summary>
-        /// <param name="symbols">The symbols to be removed keyed by SecurityType</param>
-        private void Unsubscribe(IEnumerable<Symbol> symbols)
-        {
-            lock (LockerSubscriptions)
-            {
-                var symbolsToUnsubscribe = (from symbol in symbols
-                                            where SubscribedSymbols.Contains(symbol)
-                                            select symbol).ToList();
-                if (symbolsToUnsubscribe.Count == 0)
-                    return;
-
-                Log.Trace("OandaBrokerage.Unsubscribe(): {0}", string.Join(",", symbolsToUnsubscribe.Select(x => x.Value)));
-
-                // Oanda does not allow more than a few rate streaming sessions,
-                // so we only use a single session for all currently subscribed symbols
-                var symbolsToSubscribe = SubscribedSymbols.ToList().Where(x => !symbolsToUnsubscribe.Contains(x)).ToList();
-
-                SubscribedSymbols = symbolsToSubscribe.ToHashSet();
-
-                ProcessSubscriptionRequest();
-            }
-        }
-
         /// <summary>
         /// Groups multiple subscribe/unsubscribe calls to avoid closing and reopening the streaming session on each call
         /// </summary>
         private void ProcessSubscriptionRequest()
         {
-            if (_subscriptionsPending) return;
+            if (_refreshDelay.Enabled) _refreshDelay.Stop();
+            _refreshDelay.DisposeSafely();
+            _refreshDelay = new Timer(SubscribeDelay.TotalMilliseconds);
+            _refreshDelay.Elapsed += (sender, args) =>
+            {
+                var symbolsToSubscribe = SubscribedSymbols;
```

Comment:
Not sure about this `timer`, seems there are race conditions, 🤔, we should probably have a single task started in the constructor which monitors an event that we trigger here and has to take the `lock (LockerSubscriptions)`